kube-client/src/discovery/apigroup.rs

```diff
@@ -271,4 +271,23 @@ impl ApiGroup {
         }
         None
     }
+
+    /// Comparator to help sort a collection of ApiGroups in alphabetical ordering
+    ///
+    /// This follows the default group ordering of `kubectl`.
+    ///
+    /// ```
+    /// use kube::discovery::{ApiGroup, Discovery};
+    /// # async fn wrapper() -> Result<(), Box<dyn std::error::Error>> {
+    /// # let client: kube::Client = todo!();
+    /// let discovery = Discovery::new(client).run().await?;
+    /// let groups: Vec<_> = discovery.groups().collect();
+    /// groups.sort_by(|x, y| ApiGroup::alpha_cmp(x, y));
+    /// assert_eq!(groups[0].name, ApiGroup::CORE_GROUP);
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub fn alpha_cmp(lhs: &ApiGroup, rhs: &ApiGroup) -> Ordering {
+        lhs.name.cmp(&rhs.name)
+    }
```

Comment:
@teozkr : i have implemented both your 1st and 2nd suggestions in this PR at the moment. Here is the first. This is technically a bit redundant when we expose the `.name` property of `ApiGroup` (was mistakenly private), because things like [`sort_by_cached_key`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.sort_by_cached_key) will do the job fine (and even `min_by` case in iterator work with something like [`min_by_key`](https://doc.rust-lang.org/core/iter/trait.Iterator.html#method.min_by_key) - but not really sure the iterator interface is super helpful for this as noted in parent). Do you think it's worth keeping even though people can do comparison `_by_key`?

Reply:
Oh, I thought you said the sorting logic was messier than this, with the special-casing for core groups and such.

For purely alphabetical sorting I'd agree, `sort_by_key(|x| x.name)` should be enough.

Reply:
The core group ends up being first because the empty string gets sorted before any non-empty string in an alphabetical sort :-)

Reply:
Removed `alpha_cmp`.